From 2c808f7bab71217558f04f31f3cbb9bd98f62e4a Mon Sep 17 00:00:00 2001
From: Arnd Bergmann <arnd@arndb.de>
Date: Thu, 27 Jan 2011 14:40:03 +0100
Subject: [PATCH 21/21] x25: fix blocking writes in x25_output

As pointed out by Eric Dumazet, the x25_output function
was broken by my bkl-removal patch because it removed
the ability to block while sending on a full x25 socket.

Signed-off-by: Arnd Bergmann <arnd@arndb.de>
Cc: Eric Dumazet <eric.dumazet@gmail.com>
---
 net/x25/x25_out.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

--- a/net/x25/x25_out.c
+++ b/net/x25/x25_out.c
@@ -70,7 +70,7 @@ int x25_output(struct sock *sk, struct s
 		while (skb->len > 0) {
 			release_sock(sk);
 			skbn = sock_alloc_send_skb(sk, frontlen + max_len,
-						   1, &err);
+						   noblock, &err);
 			lock_sock(sk);
 			if (!skbn) {
 				if (err == -EWOULDBLOCK && noblock){
